### PR TITLE
Add copyutil.Clone

### PR DESCRIPTION
**Pull request by pkong-ds**

### CODE REVIEW

**Comment by louischan-oursky** [comment]
You need to run go mod tidy in `/` and in `/custom_build`. And you probably cannot run it in `/custom_build`

**Comment by pkong-ds** [comment]
> And you probably cannot run it in /custom_build

You are right, can help? thanks

```
fatal: repository 'https://github.com/authgear/iamsmart/' not found
```

File: pkg/util/copyutil/clone_test.go [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+			},
+		}
+
+		m2, err := Clone(m1)
+		So(err, ShouldBeNil)
+		So(m2, ShouldResemble, m1)
+
+		// change m1
+		m1["a"] = "zzz"
+		delete(m1, "b")
```
Comment (tung2744) [inline pkg/util/copyutil/clone_test.go:24]: Try to change "c" to ensure nested objects are also deep cloned?

**Comment by pkong-ds** [comment]
Updated, thanks!

Seems still need to run `go mod tidy` in `/custom_build`; No access, maybe see if daic free to help tmr

File: pkg/util/copyutil/clone_test.go (hunk 1 above)
Comment (pkong-ds) [reply pkg/util/copyutil/clone_test.go:27]: Added [Ensure nested objects are also deep cloned in copyutil.Clone](https://github.com/authgear/authgear-server/pull/4729/commits/e1b6c0a397a31445f8e1e90d8532bd7d9a2719b5)

**Review by louischan-oursky — APPROVED** [review]

**Comment by pkong-ds** [comment]
Thanks

**Pull request merged**